### DISCUSSION TITLE
OsmAnd Web (Map portal)

### DISCUSSION THREAD

**Discussion by EugeneZmeuk** [osmandapp/OsmAnd#16567] [General]
At this discussion, we want to receive feedback and proposal about our [osmand.net/map](https://osmand.net/map) from users.
Please, read [this article](https://osmand.net/docs/user/plan-route/web) before testing and using [osmand.net/map](https://osmand.net/map): [Web portal](https://osmand.net/docs/user/plan-route/web).

<img width="1420" alt="image" src="https://user-images.githubusercontent.com/44466116/221861835-da5d0a76-5c87-4619-afe1-63f006cb3a74.png">

**Comment by mariush444** [comment]
non standard color of icons is displayed as undefined
![image](https://user-images.githubusercontent.com/66887280/221887348-5daf48b4-7d68-4ee9-8d0f-ee630a1fe6e3.png)
![image](https://user-images.githubusercontent.com/66887280/221888276-ad089da0-327d-4e8c-8581-b4efbd812fa4.png)

**Comment by alisa911** [comment]
@mariush444  Hi! Send your gpx file please or this part with point.

**Reply by mariush444** [reply]
```
  <wpt lat="43.6130724" lon="25.3940579">
    <time>2022-10-22T17:11:20Z</time>
    <name>Римски лагер Нове</name>
    <extensions>
      <osmand:color>#d00d0d</osmand:color>
      <osmand:icon>historic_castle</osmand:icon>
      <osmand:background>cirlce</osmand:background>
    </extensions>
  </wpt>
```
```
  <wpt lat="42.1333412" lon="23.15">
    <name>mini pick/name>
    <desc>40-60 min</desc>
    <extensions>
      <osmand:color>#eecc22</osmand:color>
      <osmand:icon>special_trekking</osmand:icon>
      <osmand:background>cirlce</osmand:background>
    </extensions>
  </wpt>
```
```
  <wpt lat="42.1507586" lon="23.3744716">
    <name>Пещера на Иван Рилски</name>
    <extensions>
      <osmand:color>#eecc22</osmand:color>
      <osmand:icon>natural_cave_entrance</osmand:icon>
      <osmand:background>cirlce</osmand:background>
    </extensions>
  </wpt>
```

**Comment by clownfishnemoGitHub** [comment]
Hi i am new to osmand and bought the lifetime inapp map+


Is it possible to search and view POIs at the osmand web map portal same as possible now at the osmand :)

thanks best regards

**Reply by vshcherb** [reply]
Nope not possible, we plan to enable for all users some functionality. For now web portal is useful only for OsmAnd Cloud which is OsmAnd Pro feature. Later on it will be enabled POI / Plan Route for all users

**Reply by clownfishnemoGitHub** [reply]
Is it possible to start the osmand wmp by a given mapstyle within the url ?
like https://osmand.net/map?mapstyle=offroad

**Reply by vshcherb** [reply]
Unforunately not, if you meant WMS. You can only start OsmAndServer with maps it could provided tiles as XYZ

**Reply by vshcherb** [reply]
Please note that web version doesn't have labels problem solved, so labels between tiles are cut

**Reply by Kovoschiz** [reply]
Will full functionality be available to OSM contributors, with only OSM authentication required?  This is very important for verification. Browser is much more suitable than app.

**Reply by vshcherb** [reply]
Registration will be only required for 2 kind of features we will need to provide some extra resources we don't have capacity to server such as External Route Services and for Synchronization with OsmAnd Cloud.
So registration will only work for OsmAnd Pro subscribers (OsmAnd Cloud) and it won't support OSM cause email is now used as a verification method for OsmAnd Pro

**Comment by navid-zamani** [comment]
My feedback is:

The closer it is to OsmAnd on Android, the better. Just with a smaller UI scaling so the buttons aren’t huge on a PC. :)

Frankly, I’d prefer it if was just cross-compiled to webassembly or something. (You know, like Unreal Engine and Godot are.)

With some local webstorage standing in for the Android storage. Ideally being able to sync them without any “cloud” or other questionable business practices like that. (A direct websocket connection between the web and phone process (via a QR code) would do the job just fine in case Android or browsers give trouble with just using a normal directory in the file system.)

Using the same code base would definitely save a *lot* of work. And force having a higher quality (more universal) code base too, that would make implementing it for other platforms easier.

Also, if I *have* to pay, or create an account (speaking about questionable business practices), it’s already dead in the water. Because, it being open source, free clones will pop up in seconds and dominate in some more seconds. (Paying for the server costs as an option to get rid of (metadata-free) ads is fine. Paying to finance work is too. But paying to finance the resulting *software* never is, as getting money without work is called theft/robbery aka profit.)

**Comment by drazde** [comment]
I found very usefull have different layout for different activity.

If I wuold make a hiking route I would like see the the number of the path (cai,sat for italy) like this https://hiking.waymarkedtrails.org and the difficult of that (sac scale and cai scale)

If I would make a mtb/ebike route would like see the difficult of the track and path (mtb scale)

There is no different profile for MTB and Cycling routing planning. For MTB it could be usefull have a level or scale of difficult to choose (not all people have the same skill for downhill and also for uphill)

**Reply by EugeneZmeuk** [reply]
It will be added for next releases of the OsmAnd web
Thank you for your feedback

**Comment by mindolluin** [comment]
Great that the web-interface exists! One less reason to use Google Maps.

Three things I'm currently missing:

- Ability to click on a POI/street name/feature to get information on it
- Ability to create Favorites from POI
- When creating Favorites, the address should be filled in automatically from the map data

**Reply by EugeneZmeuk** [reply]
Hello,
Thank you for your feedback.
We have all these moments in our plans.

**Reply by graham8** [reply]
POI info does pop up but I can't see a way to get rid of the info pane - it takes up a lot of space.

**Comment by graham8** [comment]
Suggest you include a link on the map to the article which describes the capabilities.

**Comment by graham8** [comment]
I couldn't get 'upload gpx to route' to work.  What are the requirements for the gpx file?

**Reply by EugeneZmeuk** [reply]
Please, send us this GPX file for testing

**Reply by graham8** [reply]
Here's one with just two waypoints.
`<?xml version="1.0" encoding="UTF-8" standalone="no"?><gpx version="1.0" creator="ITN Converter 1.94 (http://www.benichou-software.com)" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://www.topografix.com/GPX/1/0" xmlns:topografix="http://www.topografix.com/GPX/Private/TopoGrafix/0/2" xsi:schemaLocation="http://www.topografix.com/GPX/1/0 http://www.topografix.com/GPX/1/0/gpx.xsd http://www.topografix.com/GPX/Private/TopoGrafix/0/2 http://www.topografix.com/GPX/Private/TopoGrafix/0/2/topografix.xsd"><time>2023-09-14T10:27:19Z</time><bounds minlat="51.066124" minlon="1.081489" maxlat="51.151344" maxlon="1.157139"/><rte><name>TestforOsmand</name><rtept lat="51.066124" lon="1.081489"><name>29D Albert Road,Hythe CT21 6BT</name></rtept><rtept lat="51.151344" lon="1.157139"><name>Candlespur,Reece Lane,Acrise CT18 8LW</name></rtept></rte></gpx>
`

**Reply by EugeneZmeuk** [reply]
```
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<gpx version="1.0" creator="ITN Converter 1.94 (http://www.benichou-software.com)" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://www.topografix.com/GPX/1/0" xmlns:topografix="http://www.topografix.com/GPX/Private/TopoGrafix/0/2" xsi:schemaLocation="http://www.topografix.com/GPX/1/0 http://www.topografix.com/GPX/1/0/gpx.xsd http://www.topografix.com/GPX/Private/TopoGrafix/0/2 http://www.topografix.com/GPX/Private/TopoGrafix/0/2/topografix.xsd">
<time>2023-09-14T10:27:19Z</time>
<bounds minlat="51.066124" minlon="1.081489" maxlat="51.151344" maxlon="1.157139"/>
<name>TestforOsmand</name>
<wpt lat="51.066124" lon="1.081489"><name>29D Albert Road,Hythe CT21 6BT</name></wpt>
<wpt lat="51.151344" lon="1.157139"><name>Candlespur,Reece Lane,Acrise CT18 8LW</name></wpt>
</gpx> 
```

**Reply by EugeneZmeuk** [reply]
Your points have "rtept" (route points) tag. For waypoints it must be "wpt" tag for waypoints.

**Reply by EugeneZmeuk** [reply]
![image](https://github.com/osmandapp/OsmAnd/assets/44466116/f92081b0-dadf-491e-a3b2-ca1614ef420b)

**Reply by graham8** [reply]
OK, thx.  A bit counter-intuitive when importing a 'route'.

I changed the rtepts to wpts and got rid of `<rte/>` but still doesn't work.  Could you post the gpx you used, please?

**Comment by EugeneZmeuk** [comment]
[test.gpx.zip](https://github.com/osmandapp/OsmAnd/files/12609150/test.gpx.zip)

**Reply by graham8** [reply]
Sry, didn't spot you already posted it.  Anyway, yes, I can import that as a track but not as a route.

![image](https://github.com/osmandapp/OsmAnd/assets/9421800/3ab628d7-fa66-4bd4-aa34-81030ed19fb2)

**Comment by EugeneZmeuk** [comment]
Tracks menu > Local > Upload

<img width="1398" alt="image" src="https://github.com/osmandapp/OsmAnd/assets/44466116/3b8bc8c9-c06a-474e-b10a-0a95fd5e158d">

**Reply by graham8** [reply]
I don't think you have understood my point.  Yes I can upload a track, but I want to upload a **route** that is a series of waypoints on which to invoke navigation.  There seems to be the same limitation in the app so maybe I'm expecting too much.  I just thought 'Upload GPX to Route' should do what it implies, but it seems to do nothing.

**Reply by EugeneZmeuk** [reply]
Ok, I use the GPX file from "Share" route from OsmAnd app:
[route (2).gpx.zip](https://github.com/osmandapp/OsmAnd/files/12611820/route.2.gpx.zip)
<img width="701" alt="image" src="https://github.com/osmandapp/OsmAnd/assets/44466116/87c085ab-a6c7-44df-8617-ba9d8e181766">

**Comment by Dret42** [comment]
Hi, 

I would like to report a bug.

I have **a POIs collection** in my account organizaed by category folders.

I tried to upload an updated version of the same collection from an O**smAnd app backup file** (in .gpx format).

The problem: the new entries seems to be added but only the container folder is visible... 
If i try to enter the new folders to see the POIs it doesen't works.

The same is for every new items added by a backup file (POIs, tracks for example): all the imported ones are invisible!

Hoping to have done something welcome, I wish you a good day!

**Reply by EugeneZmeuk** [reply]
Hi, 
It's strange because I can backup and sync Favorites (POIs GPX format, as I understand you mean it).
What is your OsmAnd plan? (Pro or Start)

**Comment by Dret42** [comment]
My plan is the Free one.

Please note that I'm not doing sync but a simple import by file operation.

Thanks!

**Reply by EugeneZmeuk** [reply]
Would you clarify your use-case? 
Do you want to sync Favorites between several devices and check your Cloud data by the web?

**Comment by Dret42** [comment]
Ok, I try to clarify!

I'm trying to update an existing collection of POIs in my OsmAnd Web profile with a file exported from OsmAnd App. Here the steps:

1) In **OsmAnd App** > Export POIs to .gpx file
2) In **OsmAnd Web** > Import .gpx file

I'm not using the cloud Sync function.

**Reply by EugeneZmeuk** [reply]
hmm, I can export-import POIs (Favorites) (Chrome, MacOS)
<img width="1092" alt="image" src="https://github.com/osmandapp/OsmAnd/assets/44466116/4aaa9a9c-cb82-4264-a67d-c7d0d41d9ab4">

**Comment by mwerle** [comment]
Map is displayed in the local language, not in the selected "Display Language".  

I'm currently in Japan, but cannot read Japanese. Within OsmAnd, I can have (most) of the map displayed using English. The web interface does not provide a way to display the map in english.

**Reply by EugeneZmeuk** [reply]
hi, 
we will add a choice for map language.

**Reply by mwerle** [reply]
This is still not implemented - is this still on the roadmap?

**Reply by celesteking** [reply]
Need this badly. Especially useful when you're travelling. Otherwise, why would you use a map at all - you usually know the place you live in.
You could [even combine](https://www.openstreetmap.org/#map=14/42.36264/42.29543&layers=C) langs if user wants to learn the lang. Another way would be rendering them side by side "eng_name (local_name)" pulled from `name:en` and `name` keys, respectively. Or just let user select their preference providing the options "prefer English (render local if not found)", "English + Local", "Local". Another handy feature would be transliteration.

**Comment by vvazquezr** [comment]
I'm sure I'm not the first to say it, but with the functionality of following the shape of the trails when creating tracks, not just roads, if we could see satellite maps this would be the definitive track creation tool for me :)

**Reply by EugeneZmeuk** [reply]
Yes, additional map sources and satellites will be added in the future.

**Comment by wesleyverbruggen** [comment]
This has great potential! Thank you for taking time to work on this. Am I right in that it is currently not possible to select points of interest? I wish to add POI's as favourites, but I have to make a new favourite right next to POI's now.

**Reply by vshcherb** [reply]
This is coming. We don't have full interactive map, but you can already enable POI layer

**Reply by wesleyverbruggen** [reply]
Thank you! I didn't know I could enable the POI layer. I can add favourites from there, so that's great for now. Would like to thank you once again for continuing development on this. It feels way more natural to prepare trips and vacations on PC and then use it on my phone.

**Comment by moonraver** [comment]
Hello,
- do you plan any option for mappers to login with their OSM account? It would be very helpful to check our work on web with PC (not only with phone app)
- how often is web map updated? Phone app has Live update and map is updated in about 15 minutes.
Thank you

**Reply by DmitryAlexei** [reply]
Website uses monthly maps

**Comment by rachellawson** [comment]
I LOVE the idea of being able to download collections of tracks as a travel guide. I’d be very interested, though, in being able to upload an .obf to use on a map display, too. So, for example, riding dirt trails in the UK can only be done on legal routes and I have created an .obf with those routes and have a map display that highlights them, as described in https://rachelnorfolk.me/2024/oooh-i-can-ride-down-track, but I would LOVE to be able to use that map on the website map, for planning, too.


<table style="width:100%">
  <tr>
<th><img src="https://github.com/osmandapp/OsmAnd/assets/807248/52cdc346-6a17-4ab0-b723-41047fb8c56c"  width="300" height="auto"/> </td>
  </tr>
</table>

**Reply by EugeneZmeuk** [reply]
@rachellawson  It could be implemented for the web version in the future. 
P.S. I suggest you make the Custom plugin with your obf and XML files. It will be easy for distribution and installation. Write to my email eugene.kizevich@osmand.net  I think I can make it quickly.

**Comment by septatrix** [comment]
It seems like road restrictions for vehicle types are not taken into account when planning a route. For example motorcycle routes still traverse streets with `motorcycle:conditional=no @ (Sa,Su,PH)` set

**Reply by mwerle** [reply]
Relatedly, it would be nice to be able to override seasonal restrictions - for example, when planning a route in winter to ride in summer. I guess one approach could be adding a start-date to a route, which is then the date used for calculating whether roads are closed or not.

**Comment by m-u-xyz** [comment]
> Please, read [this article](https://osmand.net/docs/user/plan-route/web)

This article does no longer exist.

**Comment by anaxonda** [comment]
Is it possible to self host Osmand Web?

**Comment by lphilpot** [comment]
I'm obviously late to this discussion, having just purchased OsmAnd (and learned of this web portal). It looks like I can build routes, etc., in the portal and transfer them to my phone, right? 

Is there any documentation / tutorials on usage details? I can figure out the obvious stuff, but there are things I'm not getting -- Like how to specify the location of an intermediate waypoint after I've added it to a route. I'm sure there will be others.

Or has this been effectively abandoned?

Thanks.

**Reply by EugeneZmeuk** [reply]
Hi,

We will be updating the web portal documentation section over the next two weeks. You can find the current documentation here: https://osmand.net/docs/user/web/

**Reply by m-u-xyz** [reply]
… which currently is less than helpful, due to https://github.com/osmandapp/web/issues/1288

**Reply by lphilpot** [reply]
> Hi,
> 
> We will be updating the web portal documentation section over the next two weeks. You can find the current documentation here: https://osmand.net/docs/user/web/

OK, thanks. I'll review that.

Shortly after posting my question I realized a right-click would allow to insert an intermediate waypoint. I created, saved and imported a route into OsmAnd on my phone, but wow... It was a ~480 mile route from central Louisiana to southern Missouri, but on my phone it landed somewhere between Australia and the south pole. No idea why that happened.

It'll be awesome if we can get a reliable, workable method for specifying routes on the desktop then transferring them to a phone. Phones are great when nothing else is available, but they suck as a working environment. I like to explicitly choose my route and then use the app to help me navigate it. I don't like having an app choose a route for me.

**Reply by EugeneZmeuk** [reply]
hmm, it's a bug. Can you describe your scenario step by step?

**Reply by lphilpot** [reply]
Now that I've done it again I see the track / route is there in OA but it has a "tail" attached to it heading south. Also, keep in mind I may be doing something wrong (or suboptimal) since I'm a new user, so...  :-)

1. Created a 200 mile track from a random point in Louisiana to a random point in Arkansas (both are valid locations, though). That's the first image below.
2. Downloaded it as both a full and simplified track.
3. Transferred them to my phone.
4. As a test I loaded both tracks into Organic Maps and they imported properly.
5. In OsmAnd:
 Menu, Plan a route, Import track, (selected track), Save, then exit back to map
 Menu, Plan a route, Folders, (select track). That's the second and third images below.
(I realize either method would work, I was just testing)

Both tracks (full and simplified) behaved the same way.

Is that basically the correct procedure?

**_Images_**

The original route
<img width="644" height="828" alt="image" src="https://github.com/user-attachments/assets/570926ed-5c3d-43c7-a46a-c4f2aeeb1b1b" />

One view on my phone in OA
<img width="529" height="800" alt="image" src="https://github.com/user-attachments/assets/c4dd1f9d-ed1b-43d2-a7df-e77db2fc4a7f" />

Another view on my phone in OA
<img width="459" height="900" alt="image" src="https://github.com/user-attachments/assets/ef129724-603b-4a80-ad68-3339ef92d6ca" />

**Reply by EugeneZmeuk** [reply]
@Zirochkabila, we need to reproduce it

**Comment by lphilpot** [comment]
Question about favorites...

I've created some favorites in folders on the web portal and imported them to OA on my phone -- worked fine. But what happens if I make additional favorites in the web portal _**in the same folder**_, then import that folder again? Will only the new / additional favorites (since the last import) be added? Or will I have duplicates of my already-imported favorites? I'm hoping it's the former!  :-)

Thanks.

**Reply by lphilpot** [reply]
Through experimentation I found the answer but it raises another question. The answer is, re-importing a folder will indeed re-import ALL favorites in the folder, creating duplicates.

So the question -- How do I add additional favorites to an existing folder when importing from a web portal export file? I don't want to delete the already-imported favorites from the web portal, those are kind of my "master list". I just want to import the new ones only.

Any advice?

Thanks.

**Comment by abrusle** [comment]
Something I dislike about the web portal is that I need to create an account to do anything useful, whereas the android version doesn't require one.

**Comment by lphilpot** [comment]
If requests are being considered I'd love to be able to define individual (or selected groups of) POIs / Favorites in the web portal and then add them to my Android OSMand instances without creating duplicates on the destination device. I know I can save / import files, but a "send to device" function would be phenomenal. However, I have no idea if the underlying infrastructure can even support such a function at this point in time.

**Reply by dmpr0** [reply]
Hi! All data (except local tracks) created on the web version is saved to OsmAnd Cloud.  If you log in to the same account on your device you’ll be able to access changes and new data like tracks and favourites from the device after Sync.

**Reply by lphilpot** [reply]
Forgive me for asking but I'm away from my computer... Does that mean I can import individual (or a group of) Favorites into the device, that were created on the portal? 

Thanks

**Comment by lphilpot** [comment]
And as long as I'm requesting, three other things...

1. Road IDs

Is there an option to prioritize road's designation over its local name? For example, use "US 123" instead of "Blue Lake Road", etc.? Local names are great for, well, locals but they're typically no use when one is traveling in an unfamiliar area (i.e., precisely when a map is needed). Also sometimes there's more than one local name for the same road but there's only one official designation / ID. In the case of colocated routes, either the correct one (when routing) or something like "US 123 / WV 456" (when in Free Ride) could be used.

For example, right now I'm in the middle of a 2000+ mile road trip and I've often been in a situation where I know I need to turn onto "US 123" in a few miles. OSMand shows the prompt for an upcoming right turn onto "Blue Lake Road" instead and I have no way to confirm that's actually US 123. If there was a preference to favor official IDs over local names that would help greatly.
 
2. Current road

Having the current road ID displayed somewhere on the (Android Auto) screen would be great. In some areas, road ID signs are few and far between.

3. Free Ride

Please make Free Ride mode the default, or at least a user preference to automatically select after x seconds. It gets old having to manually select it all the time.

Thanks.
